### DISCUSSION TITLE
Using Separate Endpoints for Uplink and Downlink

### DISCUSSION THREAD

**Discussion by AmirHosseinCV** [XTLS/Xray-core#3590] [Q&A] [answered]
Is it possible to use different endpoints for uplink and downlink? I am considering the new SplitHTTP transport for this purpose. The reason for this setup is that some ISPs restrict upload speeds to unknown servers to about 0.10Mbps. Therefore, I want to use a specific CDN for uplink to avoid this limitation, but I prefer to use my own servers for downlink to reduce costs, as the CDN can be expensive. Additionally, download speeds are not an issue in my case.

**Comment by mmmray** [comment, marked as answer]
i'm aware that this feature request is often coming from iran. it's currently not possible, and if this functionality is added I think it's best to write a traffic splitter that can send upload and download to separate transports. https://github.com/XTLS/Xray-core/discussions/3482

the reason is that one CDN may support websockets, and another CDN not. it would be good to send upload via websocket and download via splithttp if the situation permits it.

(it's technically possible to perform traffic splitting today by installing nginx on the clientside and using its routing capabilities, but it's only for users who can figure it out themselves, and difficult to provide support for)
